### PR TITLE
[ALLUXIO-2160] Write unit tests for AlluxioMasterRestServiceHandler

### PR DESCRIPTION
**Pull request by schernolyas**
https://alluxio.atlassian.net/browse/ALLUXIO-2160

### CODE REVIEW

File: core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    SortedMap<String, String> entry = (SortedMap<String, String>) response.getEntity();
+    assertFalse("Properties Map must be not empty!", (entry.isEmpty()));
+  }
+
+  @Test
+  public void testGetRpcAddress() {
+    Response response = mHandler.getRpcAddress();
+    assertNotNull("Response must be not null!", response);
+    assertNotNull("Response must have a entry!", response.getEntity());
+    assertEquals("Entry must be a String!", String.class, response.getEntity().getClass());
```
Comment (aaudiber) [inline core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java]: we should consistently use either instanceof or check the equality of the classes, whichever approach you prefer.

File: core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+  public void setUp() {
+    mMaster = mock(AlluxioMaster.class);
+    mContext = mock(ServletContext.class);
+    Journal journal = mock(Journal.class);
+    mBlockMaster = new BlockMaster(journal);
+    when(mMaster.getBlockMaster()).thenReturn(mBlockMaster);
+    when(mContext.getAttribute(MasterUIWebServer.ALLUXIO_MASTER_SERVLET_RESOURCE_KEY))
+        .thenReturn(mMaster);
+    when(mMaster.getMasterAddress()).thenReturn(new InetSocketAddress("localhost", 8080));
+    when(mContext.getAttribute(MasterUIWebServer.ALLUXIO_MASTER_SERVLET_RESOURCE_KEY))
```
Comment (aaudiber) [inline core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java]: we already mocked this on line 70

File: core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+import javax.servlet.ServletContext;
+
+/**
+ * Unit tests for {@link AlluxioMasterRestServiceHandler}.
+ */
+
```
Comment (aaudiber) [inline core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java]: extra newline here

File: core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    when(mockMetricsSet.getMetrics()).thenReturn(map);
+    MetricsSystem.METRIC_REGISTRY.registerAll(mockMetricsSet);
+
+    Response response = mHandler.getMetrics();
+    assertNotNull("Response must be not null!", response);
+    assertNotNull("Response must have a entry!", response.getEntity());
+    assertTrue("Entry must be a SortedMap!", (response.getEntity() instanceof SortedMap));
+    SortedMap<String, Long> entry = (SortedMap<String, Long>) response.getEntity();
+    assertFalse("Properties Map must be not empty!", (entry.isEmpty()));
+    assertTrue("Map must contains key " + filesPinnedProperty + "!",
```
Comment (aaudiber) [inline core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java]: contains -> contain

File: core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+        entry.containsKey(filesPinnedProperty));
+    assertEquals(FILES_PINNED_TEST_VALUE, entry.get(filesPinnedProperty).longValue());
+  }
+
+  @Test
+  public void testGetStartTimeMs() {
+    Response response = mHandler.getStartTimeMs();
```
Comment (aaudiber) [inline core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java]: We should mock the mMaster's `getStartTime()` method to test a more interesting value here

File: core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java (hunk 3 above)
Comment (schernolyas) [inline core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java]: Ok. Fixed

File: core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java (hunk 4 above)
Comment (schernolyas) [inline core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java]: Ok. Fixed.

**Comment by schernolyas** [comment]
Hi @aaudiber !

see new PR  https://github.com/Alluxio/alluxio/pull/4119

**Comment by schernolyas** [comment]
closed

**Pull request closed without merging**